api/lib/DebugManager.ts

```diff
@@ -0,0 +1,321 @@
+import type winston from 'winston'
+import { customFormat } from './logger.ts'
+import archiver from 'archiver'
+import type ZWaveClient from './ZwaveClient.ts'
+import { joinPath } from './utils.ts'
+import { storeDir } from '../config/app.ts'
+import { rm, mkdir } from 'node:fs/promises'
+import { existsSync } from 'node:fs'
+
+const debugTempDir = joinPath(storeDir, '.debug-temp')
+
+export interface DebugSession {
+	startTime: Date
+	logFilePath: string
+	driverLogFilePath: string
+	transport: winston.transport
+	originalLogLevel: string
+	driverDebugTransport?: any
+}
+
+class DebugManager {
+	private session: DebugSession | null = null
+
+	/**
+	 * Initialize the debug manager by cleaning up any old temp files
+	 */
+	async init(): Promise<void> {
+		// Clean up old debug temp directory on startup
+		if (existsSync(debugTempDir)) {
+			await rm(debugTempDir, { recursive: true, force: true })
+		}
+	}
+
+	/**
+	 * Check if a debug session is active
+	 */
+	isSessionActive(): boolean {
+		return this.session !== null
+	}
+
+	/**
+	 * Start a debug capture session
+	 */
+	async startSession(
+		logContainer: winston.Container,
+		zwaveClient: ZWaveClient,
+		originalLogLevel: string,
+	): Promise<void> {
+		if (this.session) {
+			throw new Error('A debug session is already active')
+		}
+
+		// Ensure debug temp directory exists
+		await mkdir(debugTempDir, { recursive: true })
+
+		const timestamp = new Date().toISOString().replace(/[:.]/g, '-')
+		const logFilePath = joinPath(debugTempDir, `ui-logs-${timestamp}.log`)
+		const driverLogFilePath = joinPath(
+			debugTempDir,
+			`driver-logs-${timestamp}.log`,
+		)
+
+		// Create a file transport to capture UI logs
+		const { transports } = await import('winston')
+		const transport = new transports.File({
+			filename: logFilePath,
+			format: customFormat(true),
+			level: 'debug',
+		})
+
+		// Add transport to all existing loggers
+		logContainer.loggers.forEach((logger: winston.Logger) => {
+			logger.add(transport)
+			// Also set logger level to debug
+			logger.level = 'debug'
+		})
+
+		// Update driver log level to debug using updateLogConfig
+		let driverDebugTransport: any = undefined
+		if (zwaveClient.driverReady) {
+			const { createDefaultTransportFormat } = await import(
+				'@zwave-js/core/bindings/log/node'
+			)
+			const { JSONTransport } = await import(
+				'@zwave-js/log-transport-json'
+			)
```

Comment:
I see many times this now, is this a new good practice?

Reply:
@robertsLando this is backend code, right? So code splitting doesn't even happen, neither does tree shaking. Honestly, I'd move this to the top.